app/src/main/java/com/kelvin/bdaypost/birthday/add/AddBirthdayViewModel.kt

```diff
@@ -0,0 +1,97 @@
+package com.kelvin.bdaypost.birthday.add
+
+import androidx.lifecycle.*
+import com.kelvin.bdaypost.R
+import com.kelvin.bdaypost.birthday.data.ContactRepository
+import com.kelvin.bdaypost.birthday.data.model.Birthday
+import com.kelvin.bdaypost.birthday.data.model.Contact
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+class AddBirthdayViewModel(private val contactRepo: ContactRepository) : ViewModel() {
+    private val _validContactForm = MutableLiveData<ContactFormState>()
+    val validContactForm: LiveData<ContactFormState> = _validContactForm
+
+    private val _addContactResult = MutableLiveData<ContactAddedResult>()
+    val addContactResult: LiveData<ContactAddedResult> = _addContactResult
+
+    /**
+     * Validates the state of adding a new contact's bday
+     * NameError - The name must not be blank
+     * AddressError - The address must not be blank
+     * Birthday - is a drop down and default to 01/01 and can not be invalid
+     *
+     * Will update the liveData for validContactForm
+     * @return isValidForm - if the form is filled in correctly
+     */
+    fun validateContactFormState(name: String, address: String): Boolean {
+        val nameError = if (name.isBlank()) {
+            R.string.addBday_contactName_error
+        } else null
+        val addrError = if (address.isBlank()) {
+            R.string.addBday_contactAddr_error
+        } else null
+        val currContactState = ContactFormState(
+            nameError,
+            addrError,
+            nameError == null && addrError == null
```

Comment:
Can we used a named parameter here? To make it clearer what this expression means